### PR TITLE
docs(block): update block documentation

### PR DESCRIPTION
**Pull request by leohscl**
Update block documentation with constructor methods and setter methods in the main doc comment Added an example for using it to surround widgets

Solves this: https://github.com/ratatui-org/ratatui/issues/914

### CODE REVIEW

File: src/widgets/block.rs [hunk 1]
```diff
@@ -35,6 +35,30 @@ pub use title::{Position, Title};
...
+/// - [`Block::new`] creates a new [`Block`] with no border or paddings.
+///
+/// # Setter methods
+///
+/// These methods are fluent setters. They return a new [`Block`] with the specified property set.
+///
+/// - [`Block::border_style`] Defines the style of the borders.
+/// - [`Block::border_type`] Sets the symbols used to display the border (e.g. single line, double
+///   line, thick or rounded borders).
+/// - [`Block::bordered`] Create a new block with all borders shown.
```
Comment (joshka) [inline src/widgets/block.rs:49]: move thie to the constructor section

**Review by joshka — APPROVED** [review]
LGTM - 1 minor change if you have a sec

File: src/widgets/block.rs [hunk 2]
```diff
@@ -38,15 +38,15 @@ pub use title::{Position, Title};
 /// # Constructor methods
 ///
 /// - [`Block::new`] creates a new [`Block`] with no border or paddings.
+/// - [`Block::border_style`] Defines the style of the borders.
+/// - [`Block::border_type`] Sets the symbols used to display the border (e.g. single line, double
+///   line, thick or rounded borders).
```
Comment (joshka) [inline src/widgets/block.rs:43]: Can you please move these lines back down - my comment was made in error. I meant only the `bordered` function to be move - apologies.

**Review by joshka — CHANGES_REQUESTED** [review]

File: src/widgets/block.rs (hunk 2 above)
Comment (leohscl) [reply src/widgets/block.rs:43]: Yes of course !

**Review by joshka — APPROVED** [review]
Thanks for the PR :)

**Pull request merged**